### PR TITLE
create index page

### PR DESCRIPTION
**Pull request by mstacey-codes**
All relevant migrations, models, seeders, serializers, routes, and components created to make index page and categories page

### CODE REVIEW

File: client/src/components/App.js [hunk 1]
```diff
@@ -12,24 +16,24 @@ const App = (props) => {
...
 
   return (
     <Router>
       <TopBar user={currentUser} />
       <Switch>
-        <Route exact path="/">
-          <h2>Hello from react</h2>
-        </Route>
+        <Route exact path="/" component={HomePage} />
+        <Route exact path="/categories" component={CategoriesList} />
```
Comment (dovakeen118) [inline client/src/components/App.js:35]: What's the difference between the Home Page and Category List? It looks like the Home Page just displays the list, but we also have a route to see the list.

If we're adding a home page to this app, that would be a separate feature from this. This branch is already starting to get out of scope, that it involves categories index, show pages, pasta, etc. We should break each of these down into smaller, bite sized feature branches

File: client/src/components/CategoriesList.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  }, []);
+
+  const categoriesListItems = categories.map((category) => {
+    return (
+      <li key={category.id}>
+        <Link to={`/categories/${category.id}`}>{category.name}</Link>
+      </li>
+    );
+  });
+  return (
```
Comment (dovakeen118) [inline client/src/components/CategoriesList.js:33]: We should add a new empty line before this return

File: client/src/components/CategoryShow.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect } from "react";
+import PastaTile from "./PastaTile.js";
+
+const CategoryShow = (props) => {
```
Comment (dovakeen118) [inline client/src/components/CategoryShow.js:4]: The name of this "feature" or branch is called "create index page", so we really shouldn't be creating show pages here as well

File: client/src/components/CategoryShow.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  return (
+    <>
+      <div>
+        <h1>Hello from category show page {category.name}</h1>
+      </div>
+      <div>
+        <ul>{pastasList}</ul>
+      </div>
```
Comment (dovakeen118) [inline client/src/components/CategoryShow.js:47]: an `ul` doesn't need to also be in a `div`, especially if it's not used in styling/formatting

File: client/src/components/HomePage.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import CategoriesList from "./CategoriesList";
+
+const HomePage = (props) => {
+  return (
+    <>
+      <h2>Hello from Pastabilities Home Page</h2>
```
Comment (dovakeen118) [inline client/src/components/HomePage.js:7]: This component is out of scope for an index page

File: client/src/components/CategoryShow.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  } else {
+    pastasList = category.pastas.map((pasta) => {
+      return <PastaTile key={pasta.id} {...pasta} />;
+    });
+  }
+
+  return (
+    <>
+      <div>
+        <h1>Hello from category show page {category.name}</h1>
```
Comment (nwalberts) [inline client/src/components/CategoryShow.js:43]: Let's update this text to text you would want to see on the site!

File: client/src/components/CategoryShow.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    }
+  };
+
+  useEffect(() => {
+    getCategory();
+  }, []);
+
+  let pastasList;
+  if (!category.pastas[0]) {
+    pastasList = `Nothing has been added yet. Wouldn't it be great if there was a form that let you add pastas? Guess that's for another user story.`;
```
Comment (nwalberts) [inline client/src/components/CategoryShow.js:33]: as humurous as this is, let's make this production ready! what message should be displayed here? if no message, dont have this logic!

File: client/src/components/HomePage.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+      <CategoriesList />
+    </>
```
Comment (nwalberts) [inline client/src/components/HomePage.js:9]: I'll allow this for now, but this text should be replaced by "Pastabilities" or whatever formal header text you want

File: server/package.json [hunk 9]
```diff
@@ -37,5 +37,9 @@
     "passport": "^0.5.2",
     "passport-local": "^1.0.0",
     "pg": "^8.7.1"
+  },
+  "devDependencies": {
+    "dotenv": "^8.2.0",
+    "nodemon": "^2.0.7"
```
Comment (nwalberts) [inline server/package.json:43]: just make sure to get the updates I posted in slack in the future in your codebase!

File: server/src/db/seeders/PastaSeeder.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
+import { Pasta } from "../../models/index.js";
+
+class PastaSeeder {
+  static async seed() {
+    const pastasData = [
+      {
+        name: "spaghetti",
+        description: "Long Noodle good for chicken parm!",
+        categoryId: 2,
```
Comment (dovakeen118) [inline server/src/db/seeders/PastaSeeder.js:9]: This is fine to hard code `id`'s but we should ideally be querying for the exact category that we want to associate

**Review by nwalberts — CHANGES_REQUESTED** [review]

File: client/src/components/CategoryShow.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect } from "react";
+import PastaTile from "./PastaTile.js";
+
+const CategoryShow = (props) => {
+  const [category, setCategory] = useState({
+    name: "",
+    pastas: [null],
```
Comment (dovakeen118) [inline client/src/components/CategoryShow.js:7]: We should be able to just define this state for pastas as an empty array, it shouldn't have a default `null` value within the array

**Review by dovakeen118 — CHANGES_REQUESTED** [review]
You have a lot of great work here, but too much to be done on one branch. When creating new branches, we should be able to describe the feature in one sentence without using "and". Here, this feature is displaying the list of all categories, AND the show page for a category, AND the pastas associated with that category. This becomes too much to track in one feature story, and can be broken down into smaller scoped features.

1. Category Index & Show (ideally 2 separate, but I'll allow them combined here since you've put in all the hard work)
2. add in Pasta

This way, it is much easier to be able to track the history of a feature or part of the application when it's not buried with a lot of other, unrelated code in a branch

File: client/src/components/App.js (hunk 1 above)
Comment (mstacey-codes) [reply client/src/components/App.js:35]: Thanks for the comment. It was king of an on the fly addition because our Routes on the app page started to feel a little messy, but we will keep this in mind for the future.

File: server/src/db/seeders/PastaSeeder.js (hunk 10 above)
Comment (mstacey-codes) [reply server/src/db/seeders/PastaSeeder.js:9]: can use - await Category.query().findOne({name: "long"})
